lib/routing/QubitRoute.class.php

```diff
@@ -146,7 +146,7 @@ protected function generateWithTokens($parameters)
         foreach ($tokens as $token) {
             switch ($token[0]) {
                 case 'variable':
-                    if (!$optional || !isset($this->defaults[$token[3]]) || $parameters[$token[3]] != $this->defaults[$token[3]]) {
+                    if ((!$optional || !isset($this->defaults[$token[3]]) || $parameters[$token[3]] != $this->defaults[$token[3]]) && $parameters[$token[3]]) {
```

Comment:
Should we use PHP's empty() to check if $parameters[$token[3]] has anything in it at the end of this condition? Is $token[3] a string?